kernel/src/sched.rs

```diff
@@ -22,13 +22,40 @@ const KERNEL_TICK_DURATION_US: u32 = 10000;
 /// Skip re-scheduling a process if its quanta is nearly exhausted
 const MIN_QUANTA_THRESHOLD_US: u32 = 500;
 
+pub struct Processes {
+    pub process_array: &'static mut [Option<&'static process::ProcessType>],
+    pub current: Cell<usize>,
+}
```

Comment:
In general it would be nice if somehow one or all of these names indicated which scheduling policy algorithm this code implements, i.e. including `RoundRobin` would be good.

A bonus would be if this structure made some sort of interface for adding other scheduling algorithms in the future, but it is fine if that is outside of the scope of this PR.

Reply:
It sounds like what your asking for is a trait for the scheduler :) Seems like a good idea.

it would be nice to have at least two implementations (maybe a priority based scheduler in addition to round-robin?) if we're going to do it.

@bradjc do you think it makes sense as part of this pull request? Perhaps it's important here because we'd rather not change the interface to the kernel _twice_.

Reply:
I mean we only have one implementation of `ProcessType`. Having a trait so that others can experiment seems worthwhile even if we only have one scheduler algorithm. That being said, implementing the one we have is probably pretty trivial.

>@bradjc do you think it makes sense as part of this pull request? Perhaps it's important here because we'd rather not change the interface to the kernel twice.

I'm flexible, as long as the interface doesn't change twice.